pkg/build/api/types.go

```diff
@@ -328,6 +328,9 @@ type GitBuildSource struct {
 
 	// HTTPSProxy is a proxy used to reach the git repository over https
 	HTTPSProxy *string
+
+	// URLCheckTimeoutSeconds is the timeout in seconds to use when checking connectivity to the git server
+	URLCheckTimeoutSeconds *int
```

Comment:
The purpose of the git ls-remote check was that we would fail fast when you don't have connectivity to the git URL. Otherwise git would just hang forever trying to do the clone.